arviz/data/io_cmdstan.py

```diff
@@ -430,13 +436,22 @@ def log_likelihood_to_xarray(self):
             data = _unpack_ndarrays(chain_data, columns, self.dtypes)
             data_warmup = _unpack_ndarrays(chain_data_warmup, columns, self.dtypes)
         else:
-            if isinstance(log_likelihood, str):
-                log_likelihood = [log_likelihood]
-            columns = {
-                col: idx
-                for col, idx in self.posterior_columns.items()
-                if any(item == col.split(".")[0] for item in log_likelihood)
-            }
+            if isinstance(log_likelihood, dict):
+                log_lik_to_obs_name = {v: f"{k}.{{}}".format for k, v in log_likelihood.items()}
+                columns = {
+                    log_lik_to_obs_name[col_varidx[0]](col_varidx[1]): idx
+                    for col, idx in self.posterior_columns.items()
+                    if any(item == col.split(".")[0] for item in log_likelihood.values())
+                    and (col_varidx := col.split("."))
```

Comment:
```suggestion
                    and (col_varidx == col.split("."))
```

I guess here is a typo?

Reply:
It's not a typo. It always evaluates to true, the assignment expression makes `col_varidx` the result of the split. I need both parts to compose the key in the dictionary, so this seemed neater than making two calls to `split`.

Reply:
Ok, that came into python in 3.8 so our 3.7 tests are failing